### PR TITLE
Feature/sc 15657/security injections redesign

### PR DESCRIPTION
**Pull request by shanioffe**
This pull request redesigns the Security Injections collection on CLARK. The goals of the improved webpage are:
- Create a matrix to display the Secure Coding modules
- Create display for any remaining modules on CLARK
- Include a button link to each Security Injection on CLARK
- Omit the featured section since all Security Injections are already shown
- Seamless presentation on all screen sizes - monitor, desktop, tablet and mobile

Desktop:
Screenshot 2023-07-20 at 3 58 29 PM

Truncated window:
Screenshot 2023-07-20 at 3 59 13 PM

Tablet/mobile:
Screenshot 2023-07-20 at 3 59 43 PM

### CODE REVIEW

File: src/app/collection/collection-globals.scss [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // Imports from CLARK stylesheets
-@import 'vars.scss';
+@import '_vars.scss';
```
Comment (mfranz1) [inline src/app/collection/collection-globals.scss:2]:
```suggestion
@import 'vars.scss';
```

Comment (mfranz1) [reply src/app/collection/collection-globals.scss:2]: This is part of the angular upgrade

**Review by mfranz1 — CHANGES_REQUESTED** [review]
Great work on this! Just the one comment about the vars import

**Review by mfranz1 — APPROVED** [review]
Awesome!

File: src/app/collection/pages/security-injections/security-injections.component.ts [hunk 2]
```diff
@@ -1,15 +1,50 @@
...
+
+    this.navbarService.show();
+
+    this.collection = await this.collectionService.getCollection(this.abvCollection);
+
+    this.collection = await this.collectionService.getCollectionMetadata(this.abvCollection).catch(e => {
+      if (e.status === 404) {
+        this.router.navigate(['not-found']);
+      }
+    });
```
Comment (bereketsega) [inline src/app/collection/pages/security-injections/security-injections.component.ts:44]: `this.collection` is being assigned to the return value from the second assignment line 40. Is the first assignment on line 38 necessary?

**Review by bereketsega — COMMENTED** [review]
Looks great! Just had one question.

File: src/app/collection/pages/security-injections/security-injections.component.ts (hunk 2 above)
Comment (shanioffe) [reply src/app/collection/pages/security-injections/security-injections.component.ts:42]: Good catch, that was an accidental duplicate

File: src/app/collection/pages/security-injections/security-injections.component.ts [hunk 3]
```diff
@@ -1,15 +1,43 @@
...
+    @HostListener('window:resize', ['$event'])
+    onResize(event: Event) {
+    this.isDesktop = window.innerWidth >= 768;
+    }
+
+  async ngOnInit() {
+
+    this.navbarService.show();
+
+    this.collection = await this.collectionService.getCollection(this.abvCollection);
```
Comment (bereketsega) [inline src/app/collection/pages/security-injections/security-injections.component.ts:36]: I would check if the description is retrieved with `this.collectionService.getCollection()` on the security injections page. I think `this.collectionService.getCollectionMetadata()` would get the metadata which includes the description.
```suggestion
    this.collection = await this.collectionService.getCollectionMetadata(this.abvCollection);
```

**Review by jouypo23 — APPROVED** [review]
Looks good! Make sure to address BK's comments.

**Review by stef359 — APPROVED** [review]

**Review by bereketsega — APPROVED** [review]

**Pull request merged**